src/coprocessor/endpoint.rs

```diff
@@ -118,136 +94,263 @@ impl CopContext {
         }
     }
 
-    fn add_statistics_by_region(&mut self, region_id: u64, stats: &Statistics) {
-        let flow_stats = self.request_stats
+    fn add_statistics(&mut self, type_str: &str, stats: &Statistics) {
+        self.get_statistics(type_str).add_statistics(stats);
+    }
+
+    fn add_flow_stats_by_region(&mut self, region_id: u64, stats: &Statistics) {
+        let flow_stats = self.flow_stats
             .entry(region_id)
             .or_insert_with(FlowStatistics::default);
         flow_stats.add(&stats.write.flow_stats);
         flow_stats.add(&stats.data.flow_stats);
     }
 
-    fn add_scan_count(&mut self, scan_counter: &mut ScanCounter) {
-        self.scan_counter.merge(scan_counter);
-    }
-
-    fn flush_scan_count(&mut self) {
-        self.scan_counter.flush();
-    }
-}
-
-impl Context for CopContext {
-    fn on_tick(&mut self) {
-        for type_str in &[STR_REQ_TYPE_SELECT, STR_REQ_TYPE_INDEX] {
-            let this_statistics = self.get_statistics(type_str);
-            if this_statistics.count == 0 {
-                continue;
-            }
-            for (cf, details) in this_statistics.stat.details() {
-                for (tag, count) in details {
-                    COPR_SCAN_DETAILS
-                        .with_label_values(&[type_str, cf, tag])
-                        .inc_by(count as f64)
-                        .unwrap();
+    fn collect(&mut self, region_id: u64, scan_tag: &str, stats: CopStats) {
+        let CopStats {
+            stats,
+            mut scan_counter,
+        } = stats;
```

Comment:
Why not use struct `CopStats` directly?

Reply:
I prefer deconstruct it because it's more clearly.